config/webpack/demo/webpack.config.hot.js

```diff
@@ -3,12 +3,16 @@
 var _ = require("lodash"); // devDependency
 var base = require("./webpack.config.dev");
 
-// Update our own module version.
+// Clone our own module object.
 var mod = _.cloneDeep(base.module);
-// First loader needs react hot.
-mod.loaders[0].loaders = [
-  require.resolve("react-hot-loader")
-].concat(mod.loaders[0].loaders);
+
+// Update loaders array. First loader needs react-hot-loader.
+mod.loaders[0].loaders = [require.resolve("react-hot-loader")]
+  .concat(mod.loaders[0].loader ? [mod.loaders[0].loader] : [])
+  .concat(mod.loaders[0].loaders || []);
+
+// Remove single loader if any.
+mod.loaders[0].loader = null;
```

Comment:
In any non-config JS file I'd expect `var firstLoader = mod.loaders[0]` to make all this a lot easier to read.